src/components/EventModal.vue

```diff
@@ -0,0 +1,108 @@
+<template>
+  <div class="popup">
+    <div class="popup-content" id="popup-content">
+      <button class="close" @click="close">&times;</button>
+      <h2>{{ title }}</h2>
+      <br />
+      <p>{{ desc }}</p>
+      <div v-if="past">
+        <p>This event occurred on {{ dateString }}.</p>
+      </div>
+      <div v-else>
+        <div class="form-item">
+          <label>Enter your check-in code:</label>
+          <br />
+          <input
+            type="text"
+            v-model.number="code"
+            maxlength="4"
+            input-code
+            class="code"
+            placeholder="1234"
+          />
+        </div>
+        <br />
+        <button
+          class="form-item"
+          :disabled="!codeIsValid"
+          @click="submitCheckIn()"
+        >
+          Check in
+        </button>
+        <p v-if="!codeIsValid" class="error" error-code>
+          Please enter valid 4 digit code
+        </p>
+        <p v-if="badCode" class="error" error-code>
+          The code you entered was incorrect.
+        </p>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: "ListCard",
+
+  props: {
+    event: {
+      validator: prop => typeof prop === "object" || prop === null,
+      required: true
+    },
+    past: {
+      type: Boolean,
+      default: false
+    }
+  },
+
+  data() {
+    return {
+      code: null,
+      badCode: false
+    };
+  },
+
+  computed: {
+    dateString() {
+      var options = { year: "numeric", month: "short", day: "numeric" };
+      var postDate = new Date(this.date * 1000); //multiply by 1000 because unix timestamp is in seconds, whereas Date constructor takes ms
+      return postDate.toLocaleDateString("en-US", options);
+    },
+
+    //title, desc, date: return the respective property of the active event
+    title() {
+      return this.event ? this.event.title : null;
```

Comment:
Let's use optional chaining here https://babeljs.io/docs/en/babel-plugin-proposal-optional-chaining

I added a commit to this branch to support this, so pull the origin of this branch. 

so you can make this `this.event?.title`

VSCode may give you errors - as long as the code compiles on localhost it's not an issue. I can help fix the VSCode errors.